Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -41,62 +36,41 @@ export function createProtocolFilter(): Middleware {
                         const mappingString: string = baseFileName + "@" + document.fileName;
                         client.addFileAssociations(mappingString, "cpp");
                         client.sendDidChangeSettings();
-                        document = await vscode.languages.setTextDocumentLanguage(document, "cpp");
+                        // This will cause the file to be closed and reopened.
+                        void vscode.languages.setTextDocumentLanguage(document, "cpp");
+                        return;
```

Comment:
Why are we returning here now? Doesn't the rest of the code need to run?

Reply:
I believe so.